src/extension/utils.js

```diff
@@ -9,13 +9,71 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-/* eslint-disable no-console */
 
-export function loadScript(path) {
-  const script = document.createElement('script');
-  script.type = 'module';
-  script.src = chrome.runtime.getURL(path);
-  document.head.appendChild(script);
+export const DEV_URL = 'http://localhost:3000';
+export const GH_URL = 'https://github.com/';
+
+/**
+ * Loads a script as a module via <code>script</code> element in the
+ * document's <code>head</code> element.
+ * @param {string} path The script path
+ */
+export async function loadScript(path) {
+  return new Promise((resolve) => {
+    const src = chrome.runtime.getURL(path);
+    if (!document.querySelector(`script[src="${src}"]`)) {
+      const script = document.createElement('script');
+      script.type = 'module';
+      script.src = src;
+      script.onload = () => resolve();
+      document.head.appendChild(script);
+    }
+  });
+}
+
+/**
+ * Checks if a host is a valid project host.
+ * @private
+ * @param {string} host The base host
+ * @param {string} owner The owner
+ * @param {string} host The repo
+ * @returns {boolean} <code>true</code> if project host, else <code>false</code>
+ */
+export function isValidProjectHost(host, owner, repo) {
+  const [third, second, first] = host.split('.');
+  return host.endsWith(first)
+    && ['page', 'live'].includes(first)
+    && ['aem', 'hlx'].includes(second)
+    && third.endsWith(`--${repo}--${owner}`);
+}
+
+/**
+ * Returns matches from configured projects for a given tab URL.
+ * @param {Object[]} configs The project configurations
+ * @param {string} tabUrl The tab URL
+ * @returns {Promise<Object[]>} The matches
+ */
+export async function getConfigMatches(configs, tabUrl) {
+  const {
+    host: checkHost,
+  } = new URL(tabUrl);
+  // exclude disabled configs
+  configs = configs.filter((cfg) => !cfg.disabled);
```

Comment:
nit: could chain this with the call below and skip the reassignment

Reply:
Good point!